firmware/Core/Src/temperature_sensor/temperature_sensor_driver.c

```diff
@@ -0,0 +1,168 @@
+#include "main.h"
+#include "stdio.h"
+#include <stdint.h>
+
+#include "temperature_sensor/temperature_sensor.h"
+
+const uint16_t TEMP_SENSOR_device_addr =  0x91;
+const uint16_t TEMP_SENSOR_temp_register_addr  = 0x00;
+
+// write to this register to change the precision
+const uint16_t TEMP_SENSOR_config_register_addr = 0x01;
+
+// 0x00 for nine bit, 0x01 for 10 bit, 0x02 for 11 bit, 0x03 for 12 bit
+const uint8_t TEMP_SENSOR_precision = 0x01 << 5;
+
+// number used to remove fractional portions of the temperature. Default to 100 for 10 bit precision.
+uint16_t TEMP_SENSOR_precision_scaling_factor = 100;
+
+// used to see if the user has configured the temperature precision to something other than the default.
+uint8_t TEMP_SENSOR_custom_precision_set = 0;
+
+
+float TEMP_SENSOR_precision_coefficient = TEMP_SENSOR_ten_bit_precision_coefficient;
+
+Temperature_Sensor_Data_Precision_Insignificant_Bytes_t TEMP_SENSOR_precision_insignificant_bits = TEMP_SENSOR_TEN_BIT_PRECISION_INSIGNIFICANT_BYTES;
+
+
+/// @brief Reads the temperature from the STDS75DS2F and stores it in the provided variable pointer result. 
+///         Precision can be configured from 1 to 4 bits.
+///         Temperature range is -55 to 125 degrees celsius with +/- 3 degrees celsius over the whole range.
+/// @param result
+/// - Arg 0: Memory location to store the temperature result.
+/// @return 0 if successful, 1 if config register error, 2 if temp register error.
+uint8_t read_temperature(int32_t *result)
```

Comment:
Like the comment above, this should follow the naming convention you used for the other functions